src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -212,6 +214,16 @@ class SidebarPanels extends React.Component<Props, State> {
         const isMetadataSidebarRedesignEnabled = isFeatureEnabled(features, 'metadata.redesign.enabled');
         const isMetadataAiSuggestionsEnabled = isFeatureEnabled(features, 'metadata.aiSuggestions.enabled');
 
+        const availablePanels = [
+            SIDEBAR_VIEW_BOXAI,
+            SIDEBAR_VIEW_DOCGEN,
+            SIDEBAR_VIEW_SKILLS,
+            SIDEBAR_VIEW_ACTIVITY,
+            SIDEBAR_VIEW_DETAILS,
+            SIDEBAR_VIEW_METADATA,
+        ];
+        const canShowDefaultPanel = !!(defaultPanel && availablePanels.includes(defaultPanel));
```

Comment:
❓ Are all availablePanels enabled for all files? Should we include a check to ensure that the file has access to the specified panel?

For example, if SIDEBAR_VIEW_BOXAI is preserved as the default panel, but the user switches to a file or logs in with a different user who does not have access to SIDEBAR_VIEW_BOXAI, wouldn’t that break the experience? Should we handle this scenario to avoid potential issues?

Reply:
Done - this scenario is now handled and logic tested.